Marlin/src/module/stepper/cycles.h

```diff
@@ -40,177 +40,127 @@
-    #define ISR_SHAPING_BASE_CYCLES 290UL
-  #else
-    #define ISR_SHAPING_BASE_CYCLES 0UL
-  #endif
+  constexpr uint32_t isr_shaping_base_cycles = TERN0(HAS_ZV_SHAPING, 290UL);
 
   // Stepper Loop base cycles
-  #define ISR_LOOP_BASE_CYCLES 32UL
+  constexpr uint32_t isr_loop_base_cycles = 32UL;
 
   // And each stepper (start + stop pulse) takes in worst case
-  #define ISR_STEPPER_CYCLES 60UL
+  constexpr uint32_t isr_stepper_cycles = 60UL;
 
 #endif
 
 // If linear advance is disabled, the loop also handles them
-#if DISABLED(LIN_ADVANCE) && ENABLED(MIXING_EXTRUDER)
-  #define ISR_MIXING_STEPPER_CYCLES ((MIXING_STEPPERS) * (ISR_STEPPER_CYCLES))
-#else
-  #define ISR_MIXING_STEPPER_CYCLES  0UL
-#endif
+constexpr uint32_t isr_mixing_stepper_cycles = (0UL
+  #if DISABLED(LIN_ADVANCE) && ENABLED(MIXING_EXTRUDER)
+    + (MIXING_STEPPERS) * (isr_stepper_cycles)
+  #endif
+);
 
 // Add time for each stepper
-#if HAS_X_STEP
-  #define ISR_X_STEPPER_CYCLES ISR_STEPPER_CYCLES
-#endif
-#if HAS_Y_STEP
-  #define ISR_Y_STEPPER_CYCLES ISR_STEPPER_CYCLES
-#endif
-#if HAS_Z_STEP
-  #define ISR_Z_STEPPER_CYCLES ISR_STEPPER_CYCLES
-#endif
-#if HAS_I_STEP
-  #define ISR_I_STEPPER_CYCLES ISR_STEPPER_CYCLES
-#endif
-#if HAS_J_STEP
-  #define ISR_J_STEPPER_CYCLES ISR_STEPPER_CYCLES
-#endif
-#if HAS_K_STEP
-  #define ISR_K_STEPPER_CYCLES ISR_STEPPER_CYCLES
-#endif
-#if HAS_U_STEP
-  #define ISR_U_STEPPER_CYCLES ISR_STEPPER_CYCLES
-#endif
-#if HAS_V_STEP
-  #define ISR_V_STEPPER_CYCLES ISR_STEPPER_CYCLES
-#endif
-#if HAS_W_STEP
-  #define ISR_W_STEPPER_CYCLES ISR_STEPPER_CYCLES
-#endif
-#if HAS_EXTRUDERS
-  #define ISR_E_STEPPER_CYCLES ISR_STEPPER_CYCLES // E is always interpolated, even for mixing extruders
-#endif
+constexpr uint32_t isr_x_stepper_cycles = TERN0(HAS_X_STEP, isr_stepper_cycles),
+                   isr_y_stepper_cycles = TERN0(HAS_Y_STEP, isr_stepper_cycles),
+                   isr_z_stepper_cycles = TERN0(HAS_Z_STEP, isr_stepper_cycles),
+                   isr_i_stepper_cycles = TERN0(HAS_I_STEP, isr_stepper_cycles),
+                   isr_j_stepper_cycles = TERN0(HAS_J_STEP, isr_stepper_cycles),
+                   isr_k_stepper_cycles = TERN0(HAS_K_STEP, isr_stepper_cycles),
+                   isr_u_stepper_cycles = TERN0(HAS_U_STEP, isr_stepper_cycles),
+                   isr_v_stepper_cycles = TERN0(HAS_V_STEP, isr_stepper_cycles),
+                   isr_w_stepper_cycles = TERN0(HAS_W_STEP, isr_stepper_cycles),
+                   isr_e_stepper_cycles = TERN0(HAS_EXTRUDERS, isr_stepper_cycles); // E is always interpolated, even for mixing extruders
 
 // And the total minimum loop time, not including the base
-#define _PLUS_AXIS_CYCLES(A) + (ISR_##A##_STEPPER_CYCLES)
-#define MIN_ISR_LOOP_CYCLES (ISR_MIXING_STEPPER_CYCLES LOGICAL_AXIS_MAP(_PLUS_AXIS_CYCLES))
+#define _PLUS_AXIS_CYCLES(a) + (isr_##a##_stepper_cycles)
+constexpr uint32_t min_isr_loop_cycles = (isr_mixing_stepper_cycles LOGICAL_AXIS_MAP_LC(_PLUS_AXIS_CYCLES));
```

Comment:
I was about to comment that the macro will break with lower case before I noticed the `_LC`. I think it's worth considering keeping SCREAMING_CASE when switching from `#define` to `constexpr` when macros are involved, at least as a visual cue that macros are (still) involved. (There's also the git blame/code history and merge conflicts arguments but you've probably already considered them)

Reply:
Lowercase for things that aren't a `#define` makes it easy to distinguish that they are not macros and can be treated/trusted to be typed values, so that is generally preferred. We may still uppercase that axis letter, but also I want to change those individual defined values into an array indexed by axis, so stay tuned….